### ISSUE TITLE
Dockerize task processing

### ISSUE THREAD

**Issue by mlissner** [freelawproject/courtlistener#945]
Once we get Solr launched on the new server, the next piece is going to be Celery. Celery, of course is a big messy pain in the neck and I'm not at all enthusiastic about even installing it on the new server, but by moving celery and solr over there we should make our lives much better, and so we must.

**Comment by mlissner** [comment]
So this wound up to be a real boondoggle. The goal I set here was to try to take Celery and split it off into its own repo, so that if CourtListener itself ever needed task processing, we'd have a tool for doing that. Well...there's some issues with this plan, but at the heart of them is:

1. We have a lot of code that is used by both the tasks and by the front end.
2. The tasks use the same database models as the front end. 

I did a little experiment to try to create a mini django project that just copied over the models. That sorta worked, but then I started debugging a few tasks. Some were not so bad. They'd port over kind of OK with only a few dependencies. Others were terrible, relying on all kinds of code from the main CL repo. It just wasn't happening. I suppose we could somehow rewrite this so that the tasks worked directly with SQL (I guess?) or so they passed around JSON results that the caller then dealt with (I guess?), but it's all getting horrible. 

So...forget that. We can't split off the celery code from the rest of the codebase. Instead what we should do is create a Celery docker image, so that's what I'm doing now.

----

Other fun forays:

1. I thoroughly investigated switching to either rq or huey, both alternate tasks queues. I researched the hell out of both of them, but ultimately learned that neither is good enough. That took some time.

1. I'm working on upgrading Celery to the latest version. Along with doing that, I have to upgrade Django, as in #687. That's fine because it needs to be done, but damn it, it takes time. The good news is that we'll be upgraded fairly soon to a LTS release of Django. The bad news is that after this LTS our next jump requires Python 3. The next hop is going to hurt a lot, it appears.
